File: sprint1/By languages/go/lesson10code1.go
func solve(array []int) int {
    maximal_sum := 0 // Сумма пустого префикса равна 0
    for i := 0; i < len(array); i++ {
        current_sum := 0
        for j := 0; j <= i; j++ {
            current_sum += array[j]
        }
        if maximal_sum < current_sum {
            maximal_sum = current_sum
        }
    }
    return maximal_sum
}

File: sprint1/By languages/JS/lesson10code1.js
function solve(array) {
    let maximal_sum = 0; // Сумма пустого префикса равна 0
    for (let i = 0; i < array.length; i++) {
        let current_sum = array.slice(0, i + 1).reduce((a, b) => a + b, 0);
        if (maximal_sum < current_sum) {
            maximal_sum = current_sum;
        }
    }
    return maximal_sum;
}
